src/Compilation.zig

```diff
@@ -395,6 +402,21 @@ pub const AllErrors = struct {
                     for (src.notes) |note| {
                         try note.renderToStdErrInner(ttyconf, stderr_file, "note:", .Cyan, indent);
                     }
+                    if (src.reference_trace.len != 0) {
```

Comment:
We need to do one of the following things before merging:

1. Make sure that existing tools properly handle the "referenced by" format as you've created it. There are many tools that support the standard "error: file:line:column: message" format of stderr, such as IDE integration, vim plugins, terminal extensions, etc. We need to survey the landscape and make sure this format is machine-friendly.
2. Or, we can render to stderr with the "referenced by" trace using the standard "note: file:line:column: referenced here" format, when `ttyconf` has determined that the output is not a terminal.

Reply:
I would assume most tooling just looks for `path:column:line`. Rust for example emits ` --> src/main.rs:2:14` on a separate line and I'm sure they have considered the IDEs and tooling.

Reply:
I would be satisfied with this reasoning if you double-check one thing for me: try piping the compiler output to a file (not a terminal) and see if the format is different.

Reply:
Piping does result in the same error message minus the colors but one thing they have that we don't is the ability to emit errors in JSON format.